rust/src/rfb/rfb.rs

```diff
@@ -320,20 +342,31 @@ impl RFBState {
                         );
                     }
                     Err(_) => {
-                        return AppLayerResult::err();
+                        if let Some(current_transaction) = self.get_current_tx() {
+                            RFBState::set_event(current_transaction, RFBEvent::MalformedMessage);
+                        }
+                        // We failed to parse the client init.
+                        // Continue the flow but stop trying to map the protocol.
+                        self.state = parser::RFBGlobalState::Skip;
+                        return AppLayerResult::ok();
                     }
                 },
-                parser::RFBGlobalState::Message => {
-                    //todo implement RFB messages, for now we stop here
-                    return AppLayerResult::err();
-                }
-                parser::RFBGlobalState::TCServerProtocolVersion => {
-                    SCLogDebug!("Reversed traffic, expected response.");
-                    return AppLayerResult::err();
+                parser::RFBGlobalState::Skip => {
+                    // End of parseable handshake reached, skip rest of traffic
+                    return AppLayerResult::ok();
                 }
                 _ => {
-                    SCLogDebug!("Invalid state for request {}", self.state);
-                    current = b"";
+                    // We have gotten out of sync with the expected state flow.
+                    // This could happen since we use a global state (i.e. that
+                    // is used for both directions), but if traffic can not be
+                    // parsed as expected elsewhere, we might not have advanced
+                    // a state for one direction but received data in the
+                    // "unexpected" direction, causing the parser to end up
+                    // here. Let's stop trying to parse the traffic but still
+                    // accept it.
+                    SCLogDebug!("Invalid state for request: {}", self.state);
```

Comment:
Should we set an event here ?

Reply:
Can't hurt as it's indicative of some issue with basic protocol expectations.  Done.